src/v2i-hub/PortDrayagePlugin/src/WebServiceClient.cpp

```diff
@@ -0,0 +1,277 @@
+
+     // Call back for POST /inspection/
+    connect(api, &OAIDefaultApi::inspectionPostSignal, [&]() {
+        FILE_LOG(logERROR) << "Success /inspection POST";
+        loop.quit();
+    });
+    // Error call back for POST /inspection/
+    connect(api, &OAIDefaultApi::inspectionPostSignalE, [&](QNetworkReply::NetworkError error_code, QString error_str) {
+        FILE_LOG(logERROR) << "Failure /inspection POST : " << error_str.toStdString();
+        FILE_LOG(logERROR) << error_code;
+        loop.quit();
+    });
+
+    // Setup request
+    req.setVehicleId( QString::fromStdString( vehicle_id ) ) ;
+    req.setContainerId( QString::fromStdString( container_id ) );
+    req.setActionId( QString::fromStdString( action_id ) );
+
+    api->inspectionPost( req );
+    QTimer::singleShot(0, &loop, &QEventLoop::quit);
+    loop.exec();
+
+    // Poll inspection status until complete or proceed to holding
+    return pollInspectionAction( req.getActionId() );
+}
+
+void WebServiceClient::request_holding( std::string action_id ) {
+    QEventLoop loop;
+
+    // Call back for POST /inspection/
+    connect(api, &OAIDefaultApi::inspectionHoldingActionIdPostSignal, [&]() {
+        FILE_LOG(logERROR) << "Success /inspection/holding/{action_id} POST";
+        loop.quit();
+    });
+    // Error call back for POST /inspection/
+    connect(api, &OAIDefaultApi::inspectionHoldingActionIdPostSignalE, [&](QNetworkReply::NetworkError error_code, QString error_str) {
+        FILE_LOG(logERROR) << "Failure /inspection/holding/{action_id} POST : " << error_str.toStdString();
+        FILE_LOG(logERROR) << error_code;
+        loop.quit();
+    });
+
+    api->inspectionHoldingActionIdPost( QString::fromStdString(action_id) );
+    QTimer::singleShot(5000, &loop, &QEventLoop::quit);
+    loop.exec();
+
+    // Poll inspection action until complete
+    pollInspectionAction( QString::fromStdString( action_id ) );
+
+}
+
+void WebServiceClient::pollLoadingAction( QString action_id ) {
+    QEventLoop loop;
+
+     // Call back for Get /loading/{action_id}
+    connect(api, &OAIDefaultApi::loadingActionIdGetSignal, [&](OAIContainerActionStatus loading_action) {
+        FILE_LOG(logERROR) << "Success /loading/{action_id} GET : " << loading_action.asJson().toStdString();
+        current_loading_action = loading_action; 
+        loop.quit();
+    });
+    // Error call back for Get /loading/{action_id}
+    connect(api, &OAIDefaultApi::loadingActionIdGetSignalE, 
+        [&](OAIContainerActionStatus loading_action , QNetworkReply::NetworkError error_code, QString error_str) {
+        FILE_LOG(logERROR) << "Failure loading/{action_id} GET :" << error_str.toStdString();
+        FILE_LOG(logERROR) << error_code;
+        loop.quit();
+    });
+
+
+    api->loadingActionIdGet( action_id );
+    QTimer::singleShot(5000, &loop, &QEventLoop::quit);
+    loop.exec();
+
+    while ( current_loading_action.getStatus() != QString::fromStdString( "LOADED") ) {
```

Comment:
Can we update this to do...while loop since it is at least called once?

Reply:
will do